backend/app/api/api_v1/routers/tests/test_users.py

```diff
@@ -1,108 +1,65 @@
-from app.db.models.user import User
+import datetime
+from unittest.mock import patch
 
-
-def test_get_users(client, test_superuser, superuser_token_headers):
-    response = client.get("/api/v1/users", headers=superuser_token_headers)
-    assert response.status_code == 200
-    assert response.json() == [
-        {
-            "id": test_superuser.id,
-            "email": test_superuser.email,
-            "is_active": test_superuser.is_active,
-            "is_superuser": test_superuser.is_superuser,
-        }
-    ]
+from app.core.email import EmailType
+from app.db.models import User, PasswordResetToken
 
 
-def test_delete_user(client, test_superuser, test_db, superuser_token_headers):
-    response = client.delete(
-        f"/api/v1/users/{test_superuser.id}", headers=superuser_token_headers
-    )
+def test_authenticated_user_me(client, user_token_headers):
+    response = client.get("/api/v1/users/me", headers=user_token_headers)
     assert response.status_code == 200
-    assert test_db.query(User).all() == []
 
 
-def test_delete_user_not_found(client, superuser_token_headers):
-    response = client.delete("/api/v1/users/4321", headers=superuser_token_headers)
-    assert response.status_code == 404
-
+def test_unauthenticated_routes(client):
+    response = client.get("/api/v1/users/me")
+    assert response.status_code == 401
 
-def test_edit_user(client, test_superuser, superuser_token_headers):
-    new_user = {
-        "email": "newemail@email.com",
-        "is_active": False,
-        "is_superuser": True,
-        "first_name": "Joe",
-        "last_name": "Smith",
-        "password": "new_password",
-    }
 
+@patch("app.api.api_v1.routers.users.send_email")
+def test_edit_user(mock_send_email, client, test_user, user_token_headers, test_db):
     response = client.put(
-        f"/api/v1/users/{test_superuser.id}",
-        json=new_user,
-        headers=superuser_token_headers,
+        "/api/v1/users/me",
+        json={
+            "email": test_user.email,
+            "names": "New name",
+            "is_active": test_user.is_active,  # TODO prevent user from setting these flags
+            "is_superuser": test_user.is_superuser,  # TODO prevent user from setting these flags
```

Comment:
These seem pretty critical, can we open a bug ticket to make sure we don't lose this info!

Reply:
#428